tasks/src/robocorp/tasks/cli.py

```diff
@@ -9,35 +9,54 @@
 code which must be automatically logged is not imported prior the the `cli.main`
 call.
 """
+
 import os
 import sys
 import warnings
+from importlib.metadata import PackageNotFoundError, version as get_version
 from typing import List, Optional, Protocol
 
+from packaging import version as version_parser
+
 
-def inject_truststore():
-    # Use certificates from native storage (if `truststore` installed)
+def _inject_truststore():
+    # Use certificates from the native storage. (if `truststore` installed and minimum
+    #  interpreter/pip versions are met)
 
-    # _RC_TEST_USE_TRUSTSTORE env var is only used to avoid unwanted warnings in tests
-    # and should not be used for other purposes
+    # NOTE(OvidiuCode): `_RC_TEST_USE_TRUSTSTORE` env var is only used to avoid
+    #  unwanted warnings under tests, and should not be used for other purposes.
     if os.getenv("_RC_TEST_USE_TRUSTSTORE", "True").lower() in ["false", "0"]:
         return
 
-    if sys.version_info >= (3, 10):
+    try:
+        pip_version = get_version("pip")
+    except PackageNotFoundError:
+        pip_ver_ok = False
+    else:
+        pip_ver_ok = version_parser.parse(pip_version) >= version_parser.parse("23.2.1")
+
+    truststore_ssl_on = False
+    if sys.version_info >= (3, 10, 12) and pip_ver_ok:
         try:
             import truststore  # type: ignore
-
-            truststore.inject_into_ssl()
         except ModuleNotFoundError:
-            warnings.warn(
-                "Usage of the native system certificate stores can’t be enabled, ensure you have the"
-                " `robocorp-truststore` dependency installed in the environment.",
-                Warning,
-                stacklevel=2,
-            )
+            pass
+        else:
+            truststore.inject_into_ssl()
+            truststore_ssl_on = True
+
+    if not truststore_ssl_on:
```

Comment:
Only when all the steps passed, we can say that we had a successful injection. Otherwise issue that warning.